### PR TITLE
Addresses integer overflow of allocating memory sizes

### LINKED ISSUE #418: Moving cached rope frequencies to Spyre directly causes OOM
Due to current limitations in eager/inductor support for views, stack, and outer, the easiest way to compute the cached frequencies for Rope is to do it on cpu and then do this:

```py3
                self.cached_freqs[dev_idx][alpha] = torch.stack(
                    [
                        torch.cos(freqs),
                        -torch.sin(freqs),
                        torch.sin(freqs),
                        torch.cos(freqs),
                    ],
                    dim=2,
                ).view(*freqs.size(), 2, 2).to("spyre")
```

The issue is that this tensor is 131072x64x2x2, which is 64MB in DL16. When transferred to AIU, with the default settings (last dim is stick), it multiplies the size by 32, making it a 2GB tensor, which for some reason causes an OOM in the AIU memory, even though the allocator itself it's saying there's at least 96GB free. We should find out why that is happening, and also figure out if we can change the stick dimension either through making everything run properly on AIU or modifying the .to() function.

### PR DESCRIPTION
**Pull request by kcirred**
Thanks for sending a pull request! Please make sure you read the [contributing guidelines](https://github.com/torch-spyre/torch-spyre/blob/main/CONTRIBUTING.md).

#### What type of PR is this?

- [ x] bug
- [ ] feature
- [ ] documentation
- [ ] cleanup

#### What this PR does:

Changed int32_t to uint64_t when getting size so it will allow for sizes larger than 2GB and not overflow.

#### Which issue(s) this PR is related to:

Fixes #418

### CODE REVIEW

**Review by ani300 — APPROVED** [review]
lgtm if linting is done properly and the changes from #458 removed

**Review by ani300 — APPROVED** [review]
lgtm!
